### PR TITLE
Mention org and team archiving for DELETE in docs

### PR DESCRIPTION
**Pull request by justinvdm**
At the moment, `DELETE` for organization and team endpoints in the docs suggests we delete organizations and teams, when we actually archive them.

### CODE REVIEW

**Comment by justinvdm** [comment]
@rudigiesler ready for review

File: docs/http_api.rst [hunk 1]
```diff
@@ -694,9 +706,15 @@ Teams
 .. _Delete team:
 .. http:delete:: /teams/(int:team_id)
 
-    Remove a team.
+    Archive a team. The team will no longer be shown when :ref:`listing teams
```
Comment (rudigiesler) [inline docs/http_api.rst]: Should we add `by default` here too, since we have the same `archived` parameter for teams?

**Comment by rudigiesler** [comment]
Just the one comment from me.

**Comment by justinvdm** [comment]
Ready for re-review.

**Comment by rudigiesler** [comment]
:+1:

**Pull request merged**